picard/config.py

```diff
@@ -341,12 +336,51 @@ class ListOption(Option):
 setting = None
 persist = None
 
+PURGE_INTERVAL = 60000
+_thread_configs = {}
+_thread_config_lock = threading.RLock()
+
 
-def _setup(app, filename=None):
+def setup_config(app, filename=None):
     global config, setting, persist
     if filename is None:
         config = Config.from_app(app)
     else:
         config = Config.from_file(app, filename)
+    _thread_configs[threading.get_ident()] = config
     setting = config.setting
     persist = config.persist
+    QtCore.QTimer.singleShot(PURGE_INTERVAL, purge_config_instances)
+
+
+def get_config():
+    """Returns a config object for the current thread.
+
+    Config objects for threads are
+    """
```

Comment:
Overall the patch looks good, but I think we should totally get rid of global `config` (also `setting` and `persist`), because keeping those along `get_config()` is rather weird and prolly prone to errors.
It would be great if `get_config()` could work in any context (threads or not).

Reply:
get_config works in any context, you totally should use it also in the main thread. The idea to keep the main thread's config still available as config.config is really for backward compatibility. If we remove it we break a lot of plugins.

Reply:
Ah ok, it makes sense.

Reply:
And it also allows me to convert the code base step by step while still having a working system. I hope that makes it easier to update some parts of the code and test it. For starters I'll focus on those parts where we actually run threaded.